test/unittest/tests/test_session_config.cpp

```diff
@@ -0,0 +1,66 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "wt_internal.h"
+#include "wrappers/mock_session.h"
+
+/*
+ * Check a basic configuration that sets and clears a flag.
+ */
+#define FLAG_TEST(config_param, flag)                                                    \
+    TEST_CASE(config_param, "[session_config]")                                          \
+    {                                                                                    \
+        std::shared_ptr<MockSession> session_mock = MockSession::buildTestMockSession(); \
+        WT_SESSION_IMPL *session = session_mock->getWtSessionImpl();                     \
+        session->flags = 0;                                                              \
+        REQUIRE(__ut_session_config_int(session, config_param "=true") == 0);            \
+        REQUIRE(F_ISSET(session, flag));                                                 \
+        REQUIRE(__ut_session_config_int(session, config_param "=false") == 0);           \
+        REQUIRE(!F_ISSET(session, flag));                                                \
+        REQUIRE(__ut_session_config_int(session, config_param "=true") == 0);            \
+        REQUIRE(__ut_session_config_int(session, "") == 0);                              \
+        REQUIRE(F_ISSET(session, flag));                                                 \
+    }
+
+FLAG_TEST("ignore_cache_size", WT_SESSION_IGNORE_CACHE_SIZE);
+FLAG_TEST("cache_cursors", WT_SESSION_CACHE_CURSORS);
+FLAG_TEST("debug.checkpoint_fail_before_turtle_update",
+  WT_SESSION_DEBUG_CHECKPOINT_FAIL_BEFORE_TURTLE_UPDATE);
+FLAG_TEST("debug.release_evict_page", WT_SESSION_DEBUG_RELEASE_EVICT);
+
+TEST_CASE("cache_max_wait_ms", "[session_config]")
```

Comment:
How did you decide on this set of config items to test? 
Is it `all boolean flags` + `cache_max_wait_ms`, and the rest will come in a later ticket?

Reply:
This is the only set of config items that can be pulled into it's own configuration function. The decision process was:

1. Test cache_max_wait_ms, okay that was relatively easy. May as well extend it
2. Try and extend it to all of the things explicitly set in session_reconfigure. So not things that get set in __wt_txn_reconfigure. 
3. Anything that does weird things like prefetch is out of scope.